python/tank/bootstrap/configuration.py

```diff
@@ -112,3 +110,42 @@ def get_tk_instance(self, sg_user):
         log.debug("Core API code located here: %s" % inspect.getfile(tk.__class__))
 
         return tk
+
+    def _set_authenticated_user(self, user):
+        """
+        Sets the authenticated user.
+
```

Comment:
nitpick: add a comment to say that any script user configuration file will be read in and taken into account and potentially override the incoming user param?